core/src/main/kotlin/io/specmatic/core/Scenario.kt

```diff
@@ -160,10 +160,13 @@ data class Scenario(
     fun matchesStub(
         httpRequest: HttpRequest,
         serverState: Map<String, Value>,
-        mismatchMessages: MismatchMessages = DefaultMismatchMessages
+        mismatchMessages: MismatchMessages = DefaultMismatchMessages,
+        unexpectedKeyCheck: UnexpectedKeyCheck = ValidateUnexpectedKeys
```

Comment:
let this be null by default, and don't add any unexpectedKeyCheck to the resolver if it is null.